model/testing/src/icon4py/model/testing/helpers.py

```diff
@@ -66,6 +71,33 @@ def allocate_data(backend, input_data):
     return input_data
 
 
+def match_marker(markers: tuple, param: typing.Any):
+    for marker in markers:
+        if hasattr(marker, "markname"):
+            m_name = marker.markname
+        elif hasattr(marker, "name"):
+            m_name = marker.name
+        else:
+            m_name = marker
```

Comment:
What are the different cases? I propose to add the complete type annotation for the markers `tuple`, e.g. `tuple[Union[str, ???], ...]`